rustfmt.toml

```diff
@@ -1 +1,84 @@
-disable_all_formatting = true
+# Eventually this shoud be: ignore = []
+ignore = [
+       "src/blockdata",
+       "src/consensus",
+       "src/network",
+       "src/util",
+]
+
+max_width = 100
+hard_tabs = false
+tab_spaces = 4
+newline_style = "Auto"
+indent_style = "Block"
+use_small_heuristics = "Default"
+fn_call_width = 80                 # Default 60
+attr_fn_like_width = 80            # Default 70
+struct_lit_width = 50              # Default 18
+struct_variant_width = 80          # Default 35
+array_width = 80                   # Default 60
+chain_width = 80                   # Default 60
+single_line_if_else_max_width = 80 # Default 50
+wrap_comments = false
+format_code_in_doc_comments = false
+comment_width = 100             # Default 80
+normalize_comments = false
+normalize_doc_attributes = false
+license_template_path = ""
+format_strings = false
+format_macro_matchers = false
+format_macro_bodies = true
+hex_literal_case = "Preserve"
+empty_item_single_line = true
+struct_lit_single_line = true
+fn_single_line = true           # Default false
+where_single_line = false
+imports_indent = "Block"
+imports_layout = "Mixed"
+imports_granularity = "Module"     # Default "Preserve"
+group_imports = "StdExternalCrate" # Default "Preserve"
+reorder_imports = true
+reorder_modules = true
```

Comment:
I really don't like these much. Sometimes it may be nicer to logically group the imports, e.g. `pub use` at top. Sadly, there's no way to implement such logic in rustfmt.

Reply:
IMO "reliable benefit of predictable ordering" > "barely ever used benefit of expressing some logical groups, that also has to be maintained over time (which it won't)".

Reply:
+1 for what @dpc said. I do understand the desire to have 'pub use' statements separate but its just another implicit policy for devs to have to remember and like dpc said such groupings never get maintained.

TBH I'm kind of disheartened by this discussion, taking care of the import statements is, IMO, a no-brainer. If we cannot come to some basic agreement that at some stage its better to use tools than developer time it is really hard to make progress on this PR.

Lets not become a project where no changes can be made because there is always some negative in any change - implying that we get stuck with negatives we are unable to address. `rustfmt` is not perfect, nothing is, `rustfmt` is, an can always be, constantly improved but refusing to use the tools we have just stagnates us.

Reply:
Honestly, I don't care about order being lexicographically sorted much and if there's no setting to put `pub` items first (what I do in my projects) and enforcing is deemed to be too annoying, I don't care about these flags much. If there was a setting I'd want it on and don't care about the rest.

Reply:
Oh, I remembered one more annoying thing about sorting: if you rename an item then you also have to move it so instead of clear -+ diffl lines being one after another you get them separated. Not saying that it's worth turning off sorting but if there was a nicer solution, that'd be great.

Reply:
Agree on the diff lines separated thing. So just to make sure I've got it, we can leave the two 'reorder' flags on and acknowledge that it is a desired improvement to `rustfmt` to be able to separate 'pub use' statements from other imports?

Reply:
Yep, I was thinking about opening issues at their repo once we agree what we need.